packages/run-protocol/src/reserve/collateralReserve.js

```diff
@@ -0,0 +1,218 @@
+import { offerTo } from '@agoric/zoe/src/contractSupport/index.js';
+
+import { AMM_INSTANCE_KEY, makeParamManager } from './params.js';
+
+const { details: X } = assert;
+
+const makeLiquidityKeyword = keyword => `${keyword}Liquidity`;
+
+/**
+ * Collateral Reserve holds onto Collateral for the RUN protocol, and can
+ * dispense it for various purposes under governance control. It currently
+ * supports governance decisions to add liquidity to an AMM pool.
+ *
+ * This contract has the ability to mint RUN. When adding liquidity to an AMM
+ * pool, it takes a specified amount of collateral on hand, and mints new RUN to
+ * accompany it, and deposits both into an AMM pool, using the AMM's method that
+ * allows the pool balance to be determined based on the contributed funds.
+ *
+ * @param {ZCF<
+ * {
+ *   electionManager: VoteOnParamChange,
+ *   main: {
+ *     AmmInstance: ParamRecord<'instance'>,
+ *     Electorate: ParamRecord<'invitation'>,
+ *   },
+ *   governedApis: ['addLiquidityToAmmPool'],
+ * }
+ * >} zcf
+ * @param {{feeMintAccess: FeeMintAccess, initialPoserInvitation: Payment}} privateArgs
+ */
+const start = async (zcf, privateArgs) => {
+  const {
+    main: {
+      [CONTRACT_ELECTORATE]: electorateParam,
+      [AMM_INSTANCE_KEY]: ammInstance,
+    },
+  } = zcf.getTerms();
+
+  /** @type {MapStore<Brand, Keyword>} */
+  const keywordForBrand = makeStore('keywords');
+  /** @type {MapStore<Keyword, [Brand, Brand]>} */
+  const brandForKeyword = makeStore('brands');
+
+  /**
+   *
+   * @param {*} issuer
+   * @param {string} keyword unique
+   * @returns {Promise<[argumentIssuer: Issuer, liquidityIssuer: Issuer]>}
+   */
+  const ammPublicFacet = E(zcf.getZoeService()).getPublicFacet(
+    ammInstance.value,
+  );
+
+  const addIssuer = async (issuer, keyword) => {
+    const brand = await E(issuer).getBrand();
+
+    const liquidityIssuer = E(ammPublicFacet).getLiquidityIssuer(brand);
+    const liquidityBrand = await E(liquidityIssuer).getBrand();
+
+    brandForKeyword.init(keyword, [brand, liquidityBrand]);
+    keywordForBrand.init(brand, keyword);
+    return Promise.all([
+      zcf.saveIssuer(issuer, keyword),
+      zcf.saveIssuer(liquidityIssuer, makeLiquidityKeyword(keyword)),
+    ]);
+  };
+
+  const getKeywordsForBrand = brand => {
+    assert(
+      keywordForBrand.has(brand),
+      `Please call addIssuer for brand ${brand}`,
+    );
+    return keywordForBrand.get(brand);
+  };
+
+  const { initialPoserInvitation, feeMintAccess } = privateArgs;
+
+  /** a powerful object; can modify the invitation */
+  const paramManager = await makeParamManager(
+    zcf.getZoeService(),
+    ammInstance.value,
+    initialPoserInvitation,
+  );
+
+  const { wrapPublicFacet, wrapCreatorFacet, getElectorate } =
+    handleParamGovernance(zcf, paramManager);
+
+  const electorateInvAmt = getElectorate();
+  // compare invitation amount from privateArgs with electorateParam from terms
```

Comment:
it's clear that the line is comparing the amounts but not why. is this "amount" is the set of the electors?
```suggestion
  // ensure set of electors in privateArgs matches terms
```

Reply:
An invitation is a particular kind of Zoe Payment. Its amount contains all the invitation details. This is verifying that the invitation in privateArgs (an actual payment) represents the same seat in the same contract as the invitation amount (just a description) in the terms.